src/Abp/Auditing/AuditingInterceptor.cs

```diff
@@ -72,27 +78,36 @@ private void PerformAsyncAuditing(IInvocation invocation, AuditInfo auditInfo)
             if (invocation.Method.ReturnType == typeof(Task))
             {
                 invocation.ReturnValue = InternalAsyncHelper.AwaitTaskWithFinally(
-                    (Task) invocation.ReturnValue,
-                    exception => SaveAuditInfo(auditInfo, stopwatch, exception)
+                    (Task)invocation.ReturnValue,
+                    exception => SaveAuditInfo(auditInfo, stopwatch, exception, null)
                     );
             }
             else //Task<TResult>
             {
                 invocation.ReturnValue = InternalAsyncHelper.CallAwaitTaskWithFinallyAndGetResult(
                     invocation.Method.ReturnType.GenericTypeArguments[0],
                     invocation.ReturnValue,
-                    exception => SaveAuditInfo(auditInfo, stopwatch, exception)
+                    (exception, task) => SaveAuditInfo(auditInfo, stopwatch, exception, task)
                     );
             }
         }
 
-        private void SaveAuditInfo(AuditInfo auditInfo, Stopwatch stopwatch, Exception exception)
+        private void SaveAuditInfo(AuditInfo auditInfo, Stopwatch stopwatch, Exception exception, Task task)
         {
             stopwatch.Stop();
             auditInfo.Exception = exception;
             auditInfo.ExecutionDuration = Convert.ToInt32(stopwatch.Elapsed.TotalMilliseconds);
+            GetTaskResult(task, auditInfo);
```

Comment:
Not a fan of getting result from task for async execution

Reply:
i don't know how to fan, i think it may mean FillTaskResult, befause is not return any value but fill auditInfo's ReturnValues